src/parser/fragments/material.rs

```diff
@@ -1,11 +1,65 @@
 use std::any::Any;
+use std::fmt;
 
-use super::{Fragment, FragmentRef, FragmentParser, StringReference, TextureReferenceFragment};
+use super::{Fragment, FragmentParser, FragmentRef, StringReference, TextureReferenceFragment};
 
 use nom::number::complete::{le_f32, le_u32};
 use nom::sequence::tuple;
 use nom::IResult;
 
+#[derive(PartialEq)]
+pub struct TransparencyFlags(u32);
+
+impl TransparencyFlags {
+    const VISIBLE: u32 = 0x80000001;
+    const MASKED: u32 = 0x2;
+    const OPACITY: u32 = 0x4;
+    const TRANSPARENCY: u32 = 0x8;
+    const MASKED_OPAQUE: u32 = 0x10;
+    /// Found on materials using 'stumpbark' texture - unknown meaning
+    /// const UNKNOWN_STUMP_BARK: u32 = 0x540;
+
+    pub fn parse(input: &[u8]) -> IResult<&[u8], Self> {
+        let (remaining, raw_flags) = le_u32(input)?;
+        Ok((remaining, Self(raw_flags)))
+    }
+
+    fn serialize(&self) -> Vec<u8> {
+        self.0.to_le_bytes().to_vec()
+    }
+
+    /// If this material is visible.  If not, this mesh is probably used for collisions only.
+    pub fn is_visible(&self) -> bool {
+        self.0 & Self::VISIBLE == Self::VISIBLE
+    }
+
+    /// This material is masked by a key color
+    pub fn has_mask_or_transparency(&self) -> bool {
```

Comment:
Just curious what the `_or_transparency` here means. It kind of made me think this was testing both flags at first glance.

Reply:
Yeah I wasn't sure about the wording here.  I may change the wording around completely actually.  This flag is I think just to indicate this material has some kind of transparency whether using a mask or alpha channel... yeah I will rename some of this.